src/palace/manager/integration/license/opds/odl/extractor.py

```diff
@@ -1,33 +1,54 @@
 from __future__ import annotations
 
-from collections.abc import Sequence
+from collections.abc import Iterable, Sequence
+from datetime import date, datetime
+from typing import Any
+from urllib.parse import urljoin
 
 from frozendict import frozendict
 
 from palace.manager.core.exceptions import PalaceValueError
 from palace.manager.data_layer.bibliographic import BibliographicData
 from palace.manager.data_layer.circulation import CirculationData
+from palace.manager.data_layer.contributor import ContributorData
 from palace.manager.data_layer.format import FormatData
+from palace.manager.data_layer.identifier import IdentifierData
 from palace.manager.data_layer.license import LicenseData
+from palace.manager.data_layer.link import LinkData
+from palace.manager.data_layer.subject import SubjectData
 from palace.manager.integration.license.opds.odl.constants import FEEDBOOKS_AUDIO
-from palace.manager.integration.license.opds.opds2.extractor import Opds2Extractor
-from palace.manager.integration.license.opds.requests import OPDS2AuthType
 from palace.manager.opds import opds2, rwpm
 from palace.manager.opds.lcp.status import LoanStatus
 from palace.manager.opds.odl import odl
 from palace.manager.opds.odl.info import LicenseInfo, LicenseStatus
-from palace.manager.sqlalchemy.constants import MediaTypes
+from palace.manager.opds.opds2 import AcquisitionObject
+from palace.manager.sqlalchemy.constants import LinkRelations, MediaTypes
+from palace.manager.sqlalchemy.model.classification import Subject
+from palace.manager.sqlalchemy.model.contributor import Contributor
 from palace.manager.sqlalchemy.model.edition import Edition
 from palace.manager.sqlalchemy.model.licensing import (
     DeliveryMechanism,
     DeliveryMechanismTuple,
+    LicensePool,
     RightsStatus,
 )
 from palace.manager.sqlalchemy.model.resource import Hyperlink
+from palace.manager.util import first_or_default
 from palace.manager.util.log import LoggerMixin
 
 
 class OPDS2WithODLExtractor(LoggerMixin):
```

Comment:
It looks like a lot of changes here, but most of the code here is just moved over from `OPDS2Extractor` in the process of unifying the two extractor classes.